src/commands/positive.ts

```diff
@@ -275,14 +288,20 @@ export default class Positive extends Command {
         security.type === OASSecurityType.OAUTH2
       ) {
         const tokenSecurityName = security.key;
-        const tokenValue =
-          token ??
-          // eslint-disable-next-line no-await-in-loop
-          (await cli.prompt('Please provide your token', {
-            type: 'mask',
-          }));
+        let tokenValue;
+        if (!noPrompt) {
+          tokenValue =
+            token ??
+            // eslint-disable-next-line no-await-in-loop
+            (await cli.prompt('Please provide your token', {
+              type: 'mask',
+            }));
+        }
+        tokenValue = token;
+
         if (!token) {
           token = tokenValue;
+          this.log('Bearer token is undefined or empty.');
         }
```

Comment:
Same issue here as with the API key. This is why that test is suddenly calling mockPrompt twice